dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/TextWrapper/ChatToTextConverter.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.AI.ChatCompletion.TextWrapper;
+
+public class ChatToTextConverter : IChatToTextConverter
+{
+    public const string IncomingMessageStart = "<|im_start|>";
+    public const string IncomingMessageEnd = "<|im_end|>";
+    private static readonly string UserIdentifier = $"[{AuthorRole.User.Label}]";
+    private static readonly string AssistantIdentifier = $"[{AuthorRole.Assistant.Label}]";
+
+    private static readonly string[] defaultStopSequences = new[] {
+        IncomingMessageStart,
+        IncomingMessageEnd,
+        UserIdentifier,
+        AssistantIdentifier };
+
+    private sealed class ChatMessage : ChatMessageBase
+    {
+        public ChatMessage(AuthorRole authorRole, string content) : base(authorRole, content)
+        {
+        }
+    }
+
+    private sealed class ChatResultFromTextResult : IChatResult
+    {
+        private readonly ITextResult _textResults;
+
+        public ChatResultFromTextResult(ITextResult textResults)
+        {
+            this._textResults = textResults;
+        }
+
+        public ModelResult ModelResult => this._textResults.ModelResult; // TODO return ChatModelResult
```

Comment:
>// TODO return ChatModelResult [](http://example.com/codeflow?start=73&length=30)

Not sure how/if to handle this. It is potentially a breaking difference between the wrapped interface and the native chat interface.

Reply:
@RogerBarreto could you take a look at this comment?